### ISSUE TITLE
training on my own dataset

### ISSUE THREAD

**Issue by johnSmith1990** [princewang1994/TextSnake.pytorch#32]
Hi Dear Wang,
If I find out correctly,
1) **Pretraining with SynthText** section, train model using SynthText dataset.
2) In the **Training** section, we have 2 choises: a)training from scratch which train model with only TotalText dataset and b) training with pretrained model. 

Is these correct?

My question is that: I have 100 images. How can I prepare them to training? 
thanks

**Comment by princewang1994** [comment]
Hi, according to your description, your own training data are small, so the
best way is training based on SynthText or TotalText(the weight I uploaded
is based on SynthText pretrained model also) than you can try fixing some
layer of backbone network.
Preparation:
1. download the SynthText/total pretrained model
2. write your own dataset imitating dataset/total_text.py for file reading
and polygons points getting.
3. modify train_textsnake.py by adding your own datasets.

**Comment by johnSmith1990** [comment]
thanks. I labeled my data and split it to test and train and copied dataset/total_text.py for my own and edited it. but when i run train_textsnake.py it gives an error like this: link: https://file.io/9dTxb7
is it because of evaluation? other modification need? Do I use validation data?

**Comment by princewang1994** [comment]
Sorry that I always failed to access gofile.io, can you post error log on this issus directly? thx.

**Comment by princewang1994** [comment]
Ok I have saw it, that error is strange, seems that you write your own train_bullet1.py. For finding out the true error, you can print `loss`, `tr_loss`, `tcl_loss`, `sin_loss` `cos_loss`, `radii_loss` to show which one is of type `float` and we can further analyze what makes it.

**Comment by johnSmith1990** [comment]
Hi, No 
I dont write my own train file.
I only rename yours. Ok I will screen shot your idea. this error not accur for SynthText and TotalText dataset.

**Comment by johnSmith1990** [comment]
I add print lines to code. here is output for 2 run :
https://pasteboard.co/IynKOKc.jpg
 one for resume training with `TotalText `(right) and other for resume training with my `dataset`(left). in my dataset training `tcl_loss`, `cos_loss`, `sin_loss` and `radii_loss `was zero.

**Comment by princewang1994** [comment]
The most probable reason is that the dataset you create newly make unsuitble output, you should may sure `tcl` and `tr` mask is right by showing `img, train_mask, tr_mask, tcl_mask, radius_map, sin_map, cos_map` after [this](https://github.com/princewang1994/TextSnake.pytorch/blob/master/train_textsnake.py#L68) line. In  your case, these masks may be all blank, which cause the zero-loss error. If so, double check dataset you write produces right polygon.

**Comment by johnSmith1990** [comment]
yes, you right. those were blank.
but how do i correct it? 
I used `T3_v3.m` file in https://github.com/cs-chan/Total-Text-Dataset/tree/master/Annotation_tools/T3/src
for produce `.mat` gt for each image. the result was a` .mat` file for each image(as was for `TotalText` dataset). How do i produce `train_mask`, `tr_mask`, `tcl_mask`, `radius_map`, `sin_map`, `cos_map`
? If i don't do a step for producing gt files?

**Comment by princewang1994** [comment]
These masks are produced with existing code in dataset/dataload.py only
thing you should do is reading your data correctly, that is to say, create
new mydata.py in dataset/, read your own data, and call self.
get_training_data(image, polygons, image_id=image_id, image_path=image_path)
in your own dataset's __getitem__(self) function with image_id and
image_path. So you should check whether the variable image and polygons
contains right image and
coordinates.

**Comment by johnSmith1990** [comment]
**Very Very thanks for your patience and helpness.**
The problem was for `.mat` files. For `TotalText `dataset `gt ` `.mat` files **name** is `poly_gt_img#.mat` and it's `variable `is `polygt `that contain polygon and label, but `T3_v3.m` file that I used for labeling, saved gt's by **name** `gt_img#.mat` and it's matrix, `gt`. I renamed and re saved matrices and problem was solved. 
(of course when I create `mydb.py` file in dataset folder I modified `poly_gt `to `gt` and so on but not solved).
